src/main/java/mate/academy/spring/dao/impl/RoleDaoImpl.java

```diff
@@ -0,0 +1,29 @@
+package mate.academy.spring.dao.impl;
+
+import mate.academy.spring.dao.AbstractDao;
+import mate.academy.spring.dao.RoleDao;
+import mate.academy.spring.exception.DataProcessingException;
+import mate.academy.spring.model.Role;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.query.Query;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class RoleDaoImpl extends AbstractDao<Role> implements RoleDao {
+    public RoleDaoImpl(SessionFactory factory) {
+        super(factory, Role.class);
+    }
+
+    @Override
+    public Role getByName(String roleName) {
+        try (Session session = factory.openSession()) {
+            Query<Role> query = session
+                    .createQuery("select r from Role r where r.roleName = :role", Role.class);
+            query.setParameter("role", Role.RoleName.valueOf(roleName));
+            return query.uniqueResult();
+        } catch (Exception e) {
+            throw new DataProcessingException("No role with name: " + roleName, e);
```

Comment:
```suggestion
            throw new DataProcessingException("Cant get role woth name: " + roleName, e);
```

we get this exception when smth went frong while we try to execute query